kf_utils/dataservice/delete.py

```diff
@@ -119,30 +83,33 @@ def delete_entities(host, study_ids=None, safety_check=True):
     :type study_ids: list of str
     :param saftey_check: Whether to delete if resource is not at localhost
     :type safety_check: bool
-
-    :returns: Dict of errors where Keys are urls that failed delete (non 200
-    status code) and values are requests.Response objects
     """
     phrase = f"studies {pformat(study_ids)}" if study_ids else "all studies"
-    logger.info(f"Deleting {phrase} from {host}")
-
-    # Get all study ids
-    if not study_ids:
-        study_ids = yield_kfids(host, "studies", {})
-
-    # Delete entities by study id
-    errors = {}
-    for study_id in study_ids:
-        # Delete entities except "study" (it has to be handled differently)
+    print(f"Deleting {phrase} from {host}")
+
+    if study_ids:
+        # Delete entities by study id
+        for study_id in study_ids:
+            # Delete entities except "study" (it has to be handled differently)
+            for endpoint in ENDPOINTS:
+                print(f"Finding all {endpoint} from study {study_id}.")
+                kfids = yield_kfids(
+                    host, endpoint, {"study_id": study_id}, show_progress=True
+                )
+                print(f"Deleting all {endpoint} from study {study_id}.")
+                delete_kfids(host, kfids, safety_check=safety_check)
+
+            # Delete study by its kfid
+            delete_kfids(host, [study_id], safety_check=safety_check)
+    else:
+        # Delete everything
         for endpoint in ENDPOINTS:
-            params = {"study_id": study_id}
-            kfids = yield_kfids(host, endpoint, params)
-            errors.update(
-                delete_kfids(host, endpoint, kfids, safety_check=safety_check)
-            )
-        # Delete study by its kfid
-        errors.update(
-            delete_kfids(host, STUDIES, [study_id], safety_check=safety_check)
-        )
-
-    return errors
+            print(f"Finding all {endpoint}.")
+            kfids = yield_kfids(host, endpoint, {}, show_progress=True)
+            print(f"Deleting all {endpoint}.")
+            delete_kfids(host, kfids, safety_check=safety_check)
```

Comment:
I'm now getting 
```
Deleting all studies from http://localhost:5000
Finding all read-groups.
Deleting all read-groups.
oFinding all read-group-genomic-files.
Deleting all read-group-genomic-files.
oFinding all sequencing-experiments.
Deleting all sequencing-experiments.
oFinding all sequencing-experiment-genomic-files.
Deleting all sequencing-experiment-genomic-files.
oFinding all genomic-files.
Deleting all genomic-files.
```

i'm thinking that leading `o` is printed from `delete_kfids`? 
maybe something like below so that we don't try to delete nothing?
```suggestion
            print(f"Finding all {endpoint}.")
            kfids = yield_kfids(host, endpoint, {}, show_progress=True)
            if len(kfids) > 0:
                print(f"Deleting all {endpoint}.")
                delete_kfids(host, kfids, safety_check=safety_check)
            else:
                print(f"Nothing found at {endpoint}.")
```

Reply:
ok try again. you'll have to install dependencies again because I added a new one for cleaner progress output